### DISCUSSION TITLE
Usage of GraalVM compiler without JVMCI for bytecode parsing

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [oracle/graal#9467] [Q&A] [answered]
Good day.
I am working on simplifying the installation of https://github.com/beehive-lab/tornadovm library that parses byte code of passed lambdas using GrallVM compiler (compiler for short)  https://github.com/beehive-lab/TornadoVM/blob/master/tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/analyzer/CodeAnalysis.java#L76-L106  (example of initialization of StructuredGraph), converts it into an intermediate representation, and then converts it into the code of target GPU.

Currently, it uses a compiler instance registered inside of JVM as JIT for its extension. This causes some problems, the most notable being the complexity of its installation. 
It would be great to decouple the compiler instance from the JIT instance.

My question is, is it possible to do and if so could you provide simplified example of such initialization of GraalVM compiler.

**Comment by dougxc** [comment, marked as answer]
This is indeed possible and is how we run a lot of the Graal unit tests. The [GraalTutorial.java](https://github.com/oracle/graal/blob/40546be9e480206542cc87c199db8397925190d1/compiler/src/jdk.graal.compiler.test/src/jdk/graal/compiler/core/test/tutorial/GraalTutorial.java#L39) class is probably a good place to start.

**Reply by andrii0lomakin** [reply]
@dougxc Thank you !
